orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/config/WebhookConfiguration.java

```diff
@@ -124,6 +138,127 @@ public ClientHttpRequestFactory webhookRequestFactory(
     return requestFactory;
   }
 
+  /** Return if the request size is valid. Throw an exception otherwise. */
+  private void validateRequestSize(Request request, long maxRequestBytes) throws IOException {
+    if (maxRequestBytes <= 0L) {
+      return;
+    }
+
+    long headerSize = request.headers().byteCount();
+
+    // If the headers are already too big, no need to deal with the body size
+    if (headerSize > maxRequestBytes) {
```

Comment:
I do like max payload size, but having a limitation on headers seems like that would already be handled by *MOST* http servers. Albeit they all default to some non-consistent value.

Is this more from a security stand point?

Reply:
Yes, security, It's more from a quality of service / denial of service standpoint.  Like, we didn't want pipeline authors to be able to generate massive requests and bog down Spinnaker.